### PR TITLE
Add OpenGLWindow trait

### LINKED ISSUE #44: Add `OpenGLWindow` trait
See https://github.com/PistonDevelopers/glium_graphics/issues/29

This will be implemented by back-ends that supports OpenGL function pointer loading and setting a window as current for OpenGL context.

- [ ] `get_proc_address`
- [ ] `is_current`
- [ ] `make_current`

### PR DESCRIPTION
**Pull request by Potpourri**
close #44

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -25,3 +25,6 @@ git = "https://github.com/pistondevelopers/input"
 [dependencies.quack]
 git = "https://github.com/pistondevelopers/quack"
 #version = "0.0.13"
+
+[dependencies]
+libc = "0.1"
```
Comment (bvssvni) [inline Cargo.toml]: Should probably use `libc = "*"` like Gfx.

File: src/lib.rs [hunk 2]
```diff
@@ -65,6 +66,18 @@ impl<T> Window for T
...
+/// Trait for OpenGL specific operations.
+pub trait OpenGLWindow: Window {
+    /// Returns the address of an OpenGL function if it exist, else returns null pointer.
+    fn get_proc_address(&self, proc_name: &str) -> *const libc::c_void;
+
+    /// Returns true if this context is the current context.
+    fn is_current(&self) -> bool;
+
+    /// Make this context current.
+    fn make_current(&self);
```
Comment (bvssvni) [inline src/lib.rs]: Maybe `&mut self`?

File: Cargo.toml (hunk 1 above)
Comment (Potpourri) [inline Cargo.toml]: Just libc not update on crates.io yet. In git already fixed.

Comment (bvssvni) [inline Cargo.toml]: Will this work with "0.1.4"? Or does it use "0.1.0"?

File: src/lib.rs (hunk 2 above)
Comment (Potpourri) [inline src/lib.rs]: Unfortunately glium Backend trait use `&self`, it problem with glfw, but I think it can be fixed.

File: Cargo.toml (hunk 1 above)
Comment (bvssvni) [inline Cargo.toml]: Anyway, I don't expect `libc::c_void` to change, so adding `libc = "*"` should be fine.

Comment (Potpourri) [inline Cargo.toml]: It the lower bound. >= 0.1

File: src/lib.rs (hunk 2 above)
Comment (bvssvni) [inline src/lib.rs]: OK. We can use `&self` for now and see how that works out for the other back-ends.

**Comment by bvssvni** [comment]
Wait merging until libc has been updated on crates.io.

File: Cargo.toml (hunk 1 above)
Comment (bvssvni) [inline Cargo.toml]: OK.

**Comment by Potpourri** [comment]
Now rustc-serialize broken.

**Comment by Potpourri** [comment]
rustc-serialize fixed, all is well.

**Comment by bvssvni** [comment]
Nice!

**Pull request merged**